CMakeLists.txt

```
@@ -78,6 +78,7 @@ option(IREE_BUILD_SAMPLES "Builds IREE sample projects." ON)
 option(IREE_BUILD_PYTHON_BINDINGS "Builds the IREE python bindings" OFF)
 option(IREE_BUILD_TRACY "Enables building the 'iree-tracy-capture' CLI tool and includes it in runtime Python bindings." OFF)
 option(IREE_BUILD_BUNDLED_LLVM "Builds the bundled llvm-project (vs using installed)" ON)
+option(IREE_USE_SYSTEM_DEPS "Uses system libraries instead of building bundled or downloaded dependencies" OFF)
```

How about this?

```suggestion
option(IREE_USE_SYSTEM_DEPS "Uses find_package() for certain system libraries instead of downloading. Warning: may find incompatible versions." OFF)
```

We could also specify a version [range]: https://cmake.org/cmake/help/latest/command/find_package.html.

I think it's important to say _certain_ libraries, since this new option currently only controls how `nanobind` is included. We could grow that list carefully to include other packages but there are many ways to get package discovery subtly wrong (as @marbre and I are discovering on some other projects 😅), so I'd rather start with a minimal set that you need for your specific usage.

---

Added a cmake warning message